src/pool/srv_pool.c

```diff
@@ -1081,6 +1084,46 @@ fini_svc_pool(struct pool_svc *svc)
 	svc->ps_pool = NULL;
 }
 
+/*
+ * There might be some swim status inconsistency, let's check and
+ * fix it.
+ */
+static int
+pool_svc_check_node_status(struct pool_svc *svc)
+{
+	struct pool_domain	*doms;
+	int			doms_cnt;
+	int			i;
+	int			rc = 0;
+
+	doms_cnt = pool_map_find_nodes(svc->ps_pool->sp_map, PO_COMP_ID_ALL,
+				       &doms);
+	D_ASSERT(doms_cnt >= 0);
+	for (i = 0; i < doms_cnt; i++) {
+		struct swim_member_state state;
+
+		/* Only check if UPIN server becomes DEAD for now */
+		if (!(doms[i].do_comp.co_status & PO_COMP_ST_UPIN))
+			continue;
+
+		rc = crt_rank_state_get(crt_group_lookup(NULL),
+				   doms[i].do_comp.co_rank, &state);
+		if (rc != 0) {
+			D_ERROR("failed to get swim for rank %u: %d\n",
+				doms[i].do_comp.co_rank, rc);
+			return rc;
+		}
+
+		if (state.sms_status == SWIM_MEMBER_DEAD ||
+		    state.sms_status == SWIM_MEMBER_INACTIVE) {
+			D_DEBUG(DB_REBUILD, "set DOWN for %d\n",
+				doms[i].do_comp.co_rank);
+			pool_evict_rank(svc, doms[i].do_comp.co_rank);
```

Comment:
should the code here assign rc to return value of pool_evict_rank()?

Reply:
ah, right.